File: gogi/documents/extracted_document.go
package documents

import "strings"

// This file defines the ExtractedDocument struct, which represents a document that has been
// processed and extracted into sections, along with its metadata.
type ExtractedDocument struct {
	Sections []DocumentSection
	Metadata map[string]string
}

func (d *ExtractedDocument) Text() string {

	contents := make([]string, 0, len(d.Sections))

	for _, section := range d.Sections {
		contents = append(contents, section.Content)
	}

	return strings.Join(contents, "\n\n")
}

File: workers/worker_utils/documents/extracted_document.py
from pydantic import BaseModel, Field
from typing import Dict, List

from workers.worker_utils.documents.document_section import DocumentSection

class ExtractedDocument(BaseModel):
    """Parsed document with structural information"""

    sections: List[DocumentSection]
    metadata: Dict[str, str] = Field(default_factory=dict)

    @property
    def text(self) -> str:
        return "\n\n".join(s.content for s in self.sections)